packages/api/src/schemas/user/__test__/resolver.test.ts

```diff
@@ -0,0 +1,221 @@
+import * as Services from '@sf/api/services';
+import {
+  MutationCreateUserArgs,
+  MutationDeleteUserArgs,
+  MutationUpdateUserArgs,
+  QueryGetUserArgs,
+  QueryListUsersArgs,
+  User
+} from '@sf/core/user';
+import resolver from '../resolver';
+
+jest.mock('@sf/api/services');
+
+const user1: User = {
+  id: '7dccdfde-ba9b-4ee5-92c7-c186c21c8543',
+  address: 'Zboncak Cliff, East Ritamouth, 73326',
+  description: 'Description 1',
+  name: 'Elliot Breitenberg',
+  dob: '2017-11-02T05:09:39.187Z',
+  imageUrl: 'https://i.pravatar.cc/150?u=Elliot+Breitenberg',
+  createdAt: '2020-03-22T16:12:01.496Z',
+  updatedAt: '2020-07-04T04:42:43.636Z',
+};
+
+const user2: User = {
+  id: '102222fe-9e75-4e50-a229-a4dfc6831275',
+  address: 'Davis Square, Philipborough, 23884-5521',
+  description: 'Description 2',
+  name: 'Sanford Johnston',
+  dob: '2019-03-16T01:17:23.004Z',
+  imageUrl: 'https://i.pravatar.cc/150?u=Sanford+Johnston',
+  createdAt: '2019-07-05T21:17:43.124Z',
+  updatedAt: '2020-06-28T08:08:39.771Z',
+};
+
+describe('Query listUsers', () => {
+  let args: QueryListUsersArgs;
+
+  beforeEach(() => {
+    args = { pageInfo: { limit: 2, offset: 0 } };
+    jest.spyOn(Services.userService, 'findAll').mockResolvedValue({
+      items: [user1, user2],
+      offset: 0,
+      total: 50,
+    });
+  });
+
+  it('returns users list from the service', async () => {
+    const { listUsers } = resolver.Query;
+    const response = await listUsers({}, args);
+    expect(Services.userService.findAll).toBeCalledWith(args.pageInfo, args.filter);
+    expect(response).toEqual({
+      users: [user1, user2],
+      offset: 0,
+      total: 50,
+    });
+  });
+
+  it('returns users list from the service with filter', async () => {
+    const { listUsers } = resolver.Query;
+    args.filter = { name: 'filter' };
+    const response = await listUsers({}, args);
+    expect(Services.userService.findAll).toBeCalledWith(args.pageInfo, args.filter);
+    expect(response).toEqual({
+      users: [user1, user2],
+      offset: 0,
+      total: 50,
+    });
+  });
+
+  it('throws error when service function fails', async () => {
+    jest.spyOn(Services.userService, 'findAll').mockRejectedValue({});
+    const { listUsers } = resolver.Query;
+    try {
+      await listUsers({}, args);
+      expect(true).toBe(false);
+    } catch (e) {
+      expect(e).toBeDefined();
+    }
```

Comment:
Is there a way to avoid try/catch in unit-tests?

Reply:
Yeah sure. I could write the fixes here.
```suggestion
    expect(listUsers({}, args)).rejects.toThrow();
```